src/solvers/dgsem_tree/indicators.jl

```diff
@@ -212,6 +212,85 @@ end
 
 
 
+"""
+    IndicatorIDP(equations::AbstractEquations, basis;
+                 IDPPositivity=false,
+                 variables_cons=(first,),
+                 positCorrFactor=0.1)
+
+Blending indicator used for subcell shock-capturing [`VolumeIntegralShockCapturingSubcell`](@ref) including:
+- positivity limiting for conservative variables.
+
+## References
+
+- Rueda-Ramírez, Pazner, Gassner (2022)
+  "Subcell Limiting Strategies for Discontinuous Galerkin Spectral Element Methods"
+- Pazner (2020)
+  "Sparse invariant domain preserving discontinuous Galerkin methods with subcell convex limiting"
+  [arXiv:2004.08503](https://doi.org/10.1016/j.cma.2021.113876)
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct IndicatorIDP{RealT<:Real, LimitingVariablesCons, Cache} <: AbstractIndicator
+  IDPPositivity::Bool
+  variables_cons::LimitingVariablesCons # Positivity of conservative variables
+  cache::Cache
+  positCorrFactor::RealT                # Correction factor for IDPPositivity
+end
+
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+function IndicatorIDP(equations::AbstractEquations, basis;
+                      IDPPositivity=false,
+                      variables_cons=(first,),
+                      positCorrFactor=0.1)
```

Comment:
similar: is `0.1` a magic number that almost always will work? Or is it something highly problem specific - in which case, a default value might do more harm than good.

Reply:
`0.1` worked for almost every problem I simulated. So, I would like to keep this one here.

Reply:
Of course, you're the expert! Then just leave it as it is... /r